src/MongoDB.Driver.Core/Core/Operations/ChangeStreamOperation.cs

```diff
@@ -295,11 +295,13 @@ public IChangeStreamCursor<TResult> Execute(IReadBinding binding, CancellationTo
             IAsyncCursor<RawBsonDocument> cursor;
             ICursorBatchInfo cursorBatchInfo;
             BsonTimestamp initialOperationTime;
+            int maxWireVersion;
             using (var context = RetryableReadContext.Create(binding, _retryRequested, cancellationToken))
             {
                 cursor = ExecuteAggregateOperation(context, cancellationToken);
                 cursorBatchInfo = (ICursorBatchInfo)cursor;
                 initialOperationTime = GetInitialOperationTimeIfRequired(context, cursorBatchInfo);
+                maxWireVersion = context.Channel.ConnectionDescription.IsMasterResult.MaxWireVersion;
```

Comment:
`maxWireVersion` is also used in `GetInitialOperationTimeIfRequired` and it can be refactored somehow, but not sure that it's worth it